pkg/api/datavolume/store.go

```diff
@@ -0,0 +1,44 @@
+package datavolume
+
+import (
+	"fmt"
+
+	"github.com/rancher/apiserver/pkg/apierror"
+	"github.com/rancher/apiserver/pkg/types"
+	"github.com/rancher/wrangler/pkg/schemas/validation"
+
+	cdiv1beta1 "github.com/rancher/harvester/pkg/generated/controllers/cdi.kubevirt.io/v1beta1"
+)
+
+const (
+	VMKindName = "VirtualMachine"
+)
+
+type dvStore struct {
+	types.Store
+	dvCache cdiv1beta1.DataVolumeCache
+}
+
+func (s *dvStore) Delete(request *types.APIRequest, schema *types.APISchema, id string) (types.APIObject, error) {
+	if err := s.delete(request.Namespace, request.Name); err != nil {
+		return types.APIObject{}, apierror.NewAPIError(validation.ServerError, err.Error())
+	}
+	return s.Store.Delete(request, request.Schema, id)
+}
+
+func (s *dvStore) delete(namespace, name string) error {
+	dv, err := s.dvCache.Get(namespace, name)
+	if err != nil {
+		return fmt.Errorf("failed to get dv %s, %v", name, err)
+	}
+
+	if len(dv.OwnerReferences) != 0 {
+		for _, owner := range dv.OwnerReferences {
+			if owner.Kind == VMKindName {
+				return fmt.Errorf("can not delete the dataVolume %s is currently owned by the VM %s", name, owner.Name)
```

Comment:
`can not delete the volume %s which is`
We say `volume` in the UI, so we may want to use the same word in the error message for the users.